src/mcts/search.cc

```diff
@@ -2092,8 +2137,23 @@ void SearchWorker::FetchSingleNodeResult(NodeToProcess* node_to_process,
   }
   // For NN results, we need to populate policy as well as value.
   // First the value...
-  node_to_process->v = -computation.GetQVal(idx_in_computation);
-  node_to_process->d = computation.GetDVal(idx_in_computation);
+  auto v = -computation.GetQVal(idx_in_computation);
+  auto d = computation.GetDVal(idx_in_computation);
+  // Check whether root moves are from the set perspective.
+  if (params_.GetPerspective() != "none") {
+    bool root_stm = (params_.GetPerspective() == "auto")
+                        ? true
+                        : ((params_.GetPerspective() == "black") ==
```

Comment:
Here the check is on the path which is quite important for performance (unlike the code that outputs info and is only called once per few second).

So it's better not to compare strings here in the inner loop (and not fetch the parameters from the parameters dict which is also not fast), but instead cache the  values in the SearchParams class as enum or set of bools (similarly to how we do with other performance critical params).

It would also be useful to factor out this check into a separate function, and it can even be inside SearchParams, something like `GetWdlPerspective(bool is_black_to_move)`.

Reply:
Do you have a reference for this? Obviously the options could internally be represented by the (unique) first character, or is this also too expensive?

Reply:
The fetching parameters from `options_` dictionary (stored inside `params_`) is slow, that's why a copy of parameters that are referenced, is cached in `SearchParams` itself.  The string comparison itself is not slow, but it's still nice (and easy) to avoid.

What I mean is to convert string to enum and cache, similarly to what is done for the `GetHistoryFill()`. With enum it will also be possible to use switch() which may (or may not, because of required `break;`s) look nicer.

Reply:
Thanks, will change it accordingly to how `GetHistoryFill()` is handled.